packages/terra-avatar/src/variants/Avatar.jsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import memoize from 'memoize-one';
+import 'terra-base/lib/baseStyles';
+import styles from '../common/Avatar.module.scss';
+import {
+  AVATAR_VARIANTS, generateImagePlaceholder, generateImage, setColor,
+} from '../common/AvatarUtils';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Specifies the alternative text for the image.
+   */
+  alt: PropTypes.string.isRequired,
+  /**
+   * Sets the background color. Defaults to `auto`. Accepted color variants are theme specific.
+   * One of: `'auto'`, `'neutral'`, `'one'`, `'two'`, `'three'`, `'four'`, `'five'`, `'six'`, `'seven'`, `'eight'`, `'nine'`, `'ten'`.
+   */
+  color: PropTypes.oneOf(['auto', 'neutral', 'one', 'two', 'three', 'four',
+    'five', 'six', 'seven', 'eight', 'nine', 'ten']),
+  /**
+   * Value used for the hash function when color is set to `auto`. If not provided, hash function utilizes alt.
+   */
+  hashValue: PropTypes.string,
+  /**
+   * The image to display.
+   */
+  image: PropTypes.string,
+  /**
+   * One or two letters to display.
+   */
+  initials: PropTypes.string,
+  /**
+   * Whether to hide avatar from the accessiblity tree.
+   */
+  isAriaHidden: PropTypes.bool,
+  /**
+   * Whether the person is deceased. Overrides any color variant.
+   */
+  isDeceased: PropTypes.bool,
+  /**
+   * Overrides the default size.
+   */
+  size: PropTypes.string,
+};
+
+const defaultProps = {
+  color: 'auto',
+  hashValue: undefined,
+  image: undefined,
+  initials: undefined,
+  isAriaHidden: false,
+  isDeceased: false,
+  size: undefined,
+};
+
+const ImageComponent = memoize(generateImage);
```

Comment:
It's kind of weird to capitalize this variable. Makes me think it's a component definition vs. the instantiated component it is.

Reply:
Know any memoization libraries that cache more than the latest? I'll do some digging. Seems we either

1. Convert `avatar` and `facility` to class components, define `generateImagePlaceholder` and `generateImage` functions per component, then memoize.
2. Keep components functional and remove memoization.
3. Keep components functional and use memoization that caches more than the latest.

Not sure which is the best. Did I miss anything? Anyone else have input on this?

Reply:
Given that all the props for these components are easily comparable (strings/booleans), it seems like we could safely make this a PureComponent and forget memoization of the individual function at all. That would automatically implement shouldComponentUpdate for us with a shallow prop comparison. That would require a class to be defined though. React.memo() provides the same functionality for function components, but that's only available starting at 16.6 vs our 16.4.2 minimum.

Reply:
My recommendation would be to just do nothing unless performance has been deemed atrocious.